scripts/main.js

```diff
@@ -67,8 +65,16 @@
 			}
 		}
 
-		xmpp_uri_parts[0] = encodeURIComponent(xmpp_uri_parts[0]) // URL-encode the JID only
-		var xmpp_uri_encoded = xmpp_uri_parts.join("?");
+		var jid_parts = xmpp_uri_parts[0].split("@");
+
+		var local_part = jid_parts[0];
+		xmpp_params["name"] = local_part.charAt(0).toUpperCase() + local_part.slice(1);
+
+		var domain_part = jid_parts[1];
+
+		var jid_encoded = encodeURIComponent(local_part) + "@" + encodeURIComponent(domain_part)
+		xmpp_uri_parts[0] = jid_encoded
+		var xmpp_uri_encoded = xmpp_uri_parts.join("?")
```

Comment:
```suggestion
		var encoded_xmpp_uri = xmpp_uri_parts.join("?")
```
I'm not a fan of Yoda word order, but it's up to you.
If you decide to change it, it should also be done for the key in the returned object

Reply:
Me neither but see previous comment :)